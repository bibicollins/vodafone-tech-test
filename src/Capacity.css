.capacity {
  position: fixed;
    left: 350px;
    top: 200px;
  width: 100%;
  font-size: 15px;

}
.price1 {
  position: fixed;
    left: 150px;
    top: 250px;
  width: 100%;
  font-size: 15px;

}
.price2 {
  position: fixed;
  left: 350px;
  top: 250px;
  width: 100%;
  font-size: 15px;
}
