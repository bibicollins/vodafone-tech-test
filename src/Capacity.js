import React, { Component } from 'react';
import PhoneData from './data/phones.json'
import css from './Capacity.css'
export class Capacity extends Component {

  constructor(props) {
    super(props)
    this.state = {
      index: 0,
    }

this.sixtyFourCapacity = this.sixtyFourCapacity.bind(this)
this.twoHundredFiftySix = this.twoHundredFiftySix.bind(this)
}
sixtyFourCapacity() {
  this.setState({index: 0})
  console.log('sixy four was clicked')
  }
twoHundredFiftySix() {
  this.setState({index: 3})
  console.log('two hundred and fifty six was clicked')
}
  render() {
    return (
      <div >
      <button className="64" onClick={this.sixtyFourCapacity}>64GB</button>
      <button className="256" onClick={this.twoHundredFiftySix}>256GB</button>
      {PhoneData.map((phone, index)=> {
        return <div className="capacity" key={index}>
        <h4>Capacity: {phone.deviceSummary[this.state.index].memory}</h4>
        <div className="price1">
        <h4 id="oneOff">From £{phone.deviceSummary[this.state.index].priceInfo.hardwarePrice.oneOffPrice.gross} upfront cost</h4>
        </div>
        <div className="price2">
        <h4 id="month">When you pay £{phone.deviceSummary[this.state.index].priceInfo.bundlePrice.monthlyPrice.gross} a month</h4>
        </div>
        </div>

      })}
      </div>
    );
  }
}

export default Capacity;
