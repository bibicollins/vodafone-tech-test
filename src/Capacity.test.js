import React from 'react';
import PhoneData from './data/phones.json'
import { mount, shallow } from 'enzyme';
import Capacity from './Capacity';
import renderer from 'react-test-renderer';
describe('Capacity', () => {
  let capacity = mount(<Capacity />);
  it('renders the current capacity', () => {
    expect(capacity.find('div').first().text()).toContain('Capacity: 64GB')
  });

  it('changes the current monthly price if 256GB button clicked', () => {
    capacity.find('button').at(1).simulate('click');
    expect(capacity.find('div').first().text()).toContain('When you pay £46.80 a month')
    expect(capacity.find('div').first().text()).toContain('From £275 upfront cost')
  })
  it('changes the current monthly price if 64GB button clicked', () => {
    capacity.find('button').at(0).simulate('click');
    expect(capacity.find('div').first().text()).toContain('When you pay £43.20 a month')
    expect(capacity.find('div').first().text()).toContain('From £149 upfront cost')
  })
  it('renders correctly', () => {
    const tree = renderer.create(
      <Capacity />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
