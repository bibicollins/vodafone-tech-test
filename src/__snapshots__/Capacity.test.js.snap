// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Capacity renders correctly 1`] = `
<div>
  <button
    className="64"
    onClick={[Function]}
  >
    64GB
  </button>
  <button
    className="256"
    onClick={[Function]}
  >
    256GB
  </button>
  <div
    className="capacity"
  >
    <h4>
      Capacity: 
      64GB
    </h4>
    <div
      className="price1"
    >
      <h4
        id="oneOff"
      >
        From £
        149
         upfront cost
      </h4>
    </div>
    <div
      className="price2"
    >
      <h4
        id="month"
      >
        When you pay £
        43.20
         a month
      </h4>
    </div>
  </div>
</div>
`;
